File: documentation-examples-java/src/java/dev/typr/foundations/docs/duckdb/DomainType.java
package dev.typr.foundations.docs.duckdb;

import dev.typr.foundations.DuckDbType;
import dev.typr.foundations.DuckDbTypes;

@SuppressWarnings("unused")
public class DomainType {
    //start
    // Wrapper type
    public record ProductId(Long value) {}

    // Create DuckDbType from bigint
    DuckDbType<ProductId> productIdType = DuckDbTypes.bigint.bimap(ProductId::new, ProductId::value);
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/duckdb/DomainType.scala
package dev.typr.foundations.docs.duckdb
import dev.typr.scalafoundations.*
import dev.typr.scalafoundations.data.*



@SuppressWarnings(Array("unused"))
object DomainType:
  //start
  // Wrapper type
  case class ProductId(value: Long)

  // Create DuckDbType from bigint
  val productIdType: DuckDbType[ProductId] = DuckDbTypes.bigint.bimap(ProductId.apply, _.value)
  //stop
